File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/maximum-number-of-operations-to-move-ones-to-the-end.java
class Solution {
    public int maxOperations(String s) {
        int n = s.length();

        int ops = 0, po = s.charAt(0) - '0';
        for (int i = 1; i < n; i++) {
            if (s.charAt(i) == '1') {
                if (s.charAt(i) != s.charAt(i - 1)) {
                    ops += po;
                }
                po++;
            }
        }

        return (s.charAt(n - 1) == '1') ? ops : ops + po;
    }
}

File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/maximum-number-of-operations-to-move-ones-to-the-end.kt
class Solution {
    fun maxOperations(s: String): Int {
        val n: Int = s.length

        var ops: Int = 0
        var po: Int = s[0] - '0'
        for (i in 1 until n) {
            if (s[i] == '1') {
                if (s[i] != s[i - 1]) {
                    ops += po
                }
                po++
            }
        }

        return if (s[n - 1] == '1') ops else ops + po
    }
}